pubsub/subscriptions.js

```diff
@@ -13,149 +13,128 @@
 // the project specified by the GCLOUD_PROJECT environment variable. See
 // https://googlecloudplatform.github.io/gcloud-node/#/docs/google-cloud/latest/guides/authentication
 var PubSub = require('@google-cloud/pubsub');
+// [END setup]
 
-// Instantiate a pubsub client
-var pubsub = PubSub();
-// [END auth]
-
-// [START create_subscription]
-/**
- * Create a new subscription.
- *
- * @param {string} topicName Name of the topic for the new subscription.
- * @param {string} subscriptionName Name for the new subscription.
- * @param {Function} callback Callback function.
- */
 function createSubscription (topicName, subscriptionName, callback) {
-  if (!topicName) {
-    return callback(new Error('"topicName" is required!'));
-  } else if (!subscriptionName) {
-    return callback(new Error('"subscriptionName" is required!'));
-  }
+  var pubsub = PubSub();
+  var topic = pubsub.topic(topicName);
 
-  var options = {
-    reuseExisting: true
-  };
-  pubsub.subscribe(topicName, subscriptionName, options, function (err, subscription) {
+  // Get the subscription if it exists, otherwise create the subscription
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/topic?method=subscribe
+  topic.subscribe(subscriptionName, function (err, subscription, apiResponse) {
     if (err) {
       return callback(err);
     }
 
     console.log('Created subscription %s to topic %s', subscriptionName, topicName);
-    return callback(null, subscription);
+    return callback(null, subscription, apiResponse);
   });
 }
-// [END create_subscription]
-
-// [START delete_subscription]
-/**
- * Delete a subscription.
- *
- * @param {string} subscriptionName Name of the subscription to delete.
- * @param {Function} callback The callback function.
- */
-function deleteSubscription (subscriptionName, callback) {
-  if (!subscriptionName) {
-    return callback(new Error('"subscriptionName" is required!'));
-  }
 
+function deleteSubscription (subscriptionName, callback) {
+  var pubsub = PubSub();
   var subscription = pubsub.subscription(subscriptionName);
 
   // Delete the subscription
-  subscription.delete(function (err) {
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/subscription?method=delete
+  subscription.delete(function (err, apiResponse) {
     if (err) {
       return callback(err);
     }
 
     console.log('Deleted subscription: %s', subscriptionName);
-    return callback(null);
+    return callback(null, apiResponse);
+  });
+}
+
+function getSubscriptionMetadata (subscriptionName, callback) {
+  var pubsub = PubSub();
+  var subscription = pubsub.subscription(subscriptionName);
+
+  // Get the metadata for the specified subscription
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/subscription?method=getMetadata
+  subscription.getMetadata(function (err, metadata) {
```

Comment:
There isn't actually an `apiResponse`, either the gcloud-node docs are wrong or there's a bug.